docs/DeveloperGuide.adoc

```diff
@@ -96,64 +112,413 @@ The `UI` component,
+
+.Structure of the Contact Storage Component
+image::ContactStorageClassDiagram.png[]
+
+*API* : link:{repoURL}/src/main/java/tagline/storage/note/ContactBookStorage.java[`ContactBookStorage.java`]
+
+The `ContactStorage` component,
+
 * can save the Address Book data in json format and read it back.
 
+[[Design-NoteStorage]]
+==== Note Storage component
+
+.Structure of the Note Storage Component
+image::NoteStorageClassDiagram.png[]
+
+*API* : link:{repoURL}/src/main/java/tagline/storage/note/NoteBookStorage.java[`NoteBookStorage.java`]
+
+The `NoteStorage` component,
+
+* can save `Note` objects in json format and read it back.
+* can save `NoteIdCounter` state in json format and read it back.
+* can save the Note Book data in json format and read it back.
+
 [[Design-Commons]]
 === Common classes
 
-Classes used by multiple components are in the `seedu.addressbook.commons` package.
+Classes used by multiple components are in the `tagline.commons` package.
 
 == Implementation
 
 This section describes some noteworthy details on how certain features are implemented.
 
-// tag::undoredo[]
+=== [Proposed] Chatbot prompting feature
+
+==== Description
+
+When the user enters an incomplete command, the command could be missing only a few compulsory fields. Instead of forcing the user to edit the command entirely, TagLine will prompt the user for further details instead.
+
+At this point, the user may abort the command or provide the requested details. When all details are provided, the command is executed.
+
+==== Proposed Implementation
+
+The prompting mechanism uses `Prompt` objects to represent individual queries for additional information. A list of `Prompt` objects is used to pass information between the `Logic` and `Ui` components. `Prompt` implements the following operations:
+
+`Logic` side:
+
+* `Prompt#setPrefix()` -- Sets the prefix of the queried data
+* `Prompt#setQuestion()` -- Sets the prompt to be displayed to the user
+* `Prompt#getResponse()` -- Gets the prompt response from the user
+
+`Ui` side:
+
+* `Prompt#getQuestion()` -- Gets the prompt to be displayed to the user
+* `Prompt#setResponse()` -- Sets the response from the user
+
+Given below is an example scenario where the user command has missing compulsory fields.
+
+Step 1: The `Ui` passes the user's command to `Logic`, which finds one or more missing compulsory fields. For each missing field, it creates a new `Prompt` object with a question. Then it throws a `ParseException` containing the list of `Prompt` objects.
+
+image::UserPromptSequenceDiagramStep1.png[]
+
+Step 2: The `Ui` receives the list of `Prompt` objects. For each `Prompt`, it retrieves the question and obtains the corresponding user feedback.
+
+image::UserPromptSequenceDiagramStep2.png[]
+
+Step 3: The `Ui` passes the original command, together with the processed `Prompt` objects, back to `Logic`. `Logic` then executes the corrected command.
+
+image::UserPromptSequenceDiagramStep3.png[]
```

Comment:
Might be better to merge all 3 diagrams together to make it more obvious that the process is continuous. The 3  steps can then be listed at the bottom of the diagram. As of now, it is quite obvious that the 3 diagrams should be viewed together but they look incomplete when viewed in isolation.